pow/src/main/java/org/ethereum/beacon/pow/DepositContract.java

```diff
@@ -15,9 +15,9 @@
 
   List<Deposit> peekDeposits(int count, Eth1Data eth1Data, UInt64 fromIndex);
 
-  Eth1Data getEth1Data(long followDistance);
+  Eth1Data getEth1DataByDistance(long distanceFromHead);
 
-  Optional<Hash32> getDepositRoot(Hash32 blockHash);
+  Optional<Eth1Data> getEth1DataByBlockHash(Hash32 blockHash);
```

Comment:
I've added `hasDepositRoot(Hash32 blockHash, Hash32, depositRoot)` method. Since many deposit roots may exist in a bock

Reply:
Yep, sounds reasonable.